.github/workflows/test.yml

```diff
@@ -46,7 +46,11 @@ jobs:
           go test "${args[@]}" -timeout 800s ./...
 
   test-tip:
-    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [ubuntu-latest, windows-2019]
```

Comment:
Somewhat unrelated to this PR, but should we also pin the version of Ubuntu, to avoid any issues when GH upgrades it? Same for the the other jobs.

Reply:
Hmm, wouldn't it be better to surface any issue earlier rather than later? We can always pin it if it stars to fail, but we'd at least know there is a problem.

Reply:
We've had issues with this before in some repo, I forget where now. And it's why Windows is pinned. I'd rather avoid surprises and possibly difficult to troubleshoot failures by pinning everything, just as we do with all dependencies, and updating manually on our schedule.

We could also add a `next` job, like we do with `test-tip`, and make any failures optional, so that they don't block PRs, but we would notice if they fail.